File: frontend/js/tile.js
class Tile {
  constructor(colour=0, empty=true) {
    this.colour = colour;
    this.empty = empty;
  }

  // show() {
  //   rectMode(CENTER);
  //   let centerX = this.side / 2 + this.pos.x;
  //   let centerY = this.side / 2 + this.pos.y;
  //   fill(this.colour);
  //   rect(centerX, centerY, this.side - this.padding, this.side - this.padding, this.radius);
  // }

  update(colour) {
    this.colour = colour;
    if (colour === 0) {
      this.empty = true;
    } else {
      this.empty = false;
    }
  }
}

File: frontend/tile.py
from dataclasses import dataclass, field

from pygame import Color


@dataclass
class Tile:
    color: Color = field(default_factory=lambda: Color(40, 40, 40))
    empty: bool = True

    def update(self, color: Color):
        self.color = color
        self.empty = False

    def clear(self):
        self.update(Color(40, 40, 40))
        self.empty = True
